Forge.TreeWalker/src/TreeSchemaValidator.cs

```diff
@@ -0,0 +1,275 @@
+        }
+
+        public async Task<Tuple<bool, IList<string>>> ValidateSchemaString(string schema, string rules)
+        {
+            return await Task.Run(() => Validate(new List<object> { schema }, rules));
+        }
+
+        public async Task<Tuple<bool, IList<string>>> ValidateSchemasString(string schemas, string rules)
+        {
+            return await Task.Run(() => Validate(new List<object> { schemas }, ForgeSchemaValidationRules));
+        }
+
+        public async Task<Tuple<bool, IList<string>>> ValidateSchemaInPath(string path, string rules)
+        {
+            return await GetschemaFromPathAndValidate(path, rules);
+        }
+
+        public async Task<Tuple<bool, IList<string>>> ValidateMultipleSchemasInPath(string path, string rules)
+        {
+            return await GetAllSchemasInDirectoryAndValidate(path, rules);
+        }
+
+        public async Task<Tuple<bool, IList<string>>> ValidateSchema(ForgeTree schema, JSchema rules)
+        {
+            return await Task.Run(() => Validate(new List<object> { schema }, ForgeSchemaValidationRules));
+        }
+
+        public async Task<Tuple<bool, IList<string>>> ValidateSchemas(IList<ForgeTree> schemas, JSchema rules)
+        {
+            return await Task.Run(() => Validate(ConvertSchemasToSchemaList(schemas), rules));
+        }
+
+        public async Task<Tuple<bool, IList<string>>> ValidateSchemasString(string schemas, JSchema rules)
+        {
+            return await Task.Run(() => Validate(new List<object> { schemas }, ForgeSchemaValidationRules));
+        }
+
+        public async Task<Tuple<bool, IList<string>>> ValidateSchemaInPath(string path, JSchema rules)
+        {
+            return await GetschemaFromPathAndValidate(path, rules);
+        }
+
+        public async Task<Tuple<bool, IList<string>>> ValidateMultipleSchemasInPath(string path, JSchema rules)
+        {
+            return await GetAllSchemasInDirectoryAndValidate(path, rules);
+        }
+
+        public async Task<Tuple<bool, IList<string>>> ValidateSchemaString(string schema, JSchema rules)
+        {
+            return await Task.Run(() => Validate(new List<object> { schema }, rules));
+        }
+
+        private static List<object> ConvertStringToSchemaList(string schemas, out bool success)
+        {
+            try
+            {
+                MemoryStream ms = new MemoryStream(Encoding.UTF8.GetBytes(schemas));
+                DataContractJsonSerializer ser = new DataContractJsonSerializer(typeof(List<ForgeTree>));
+                var schemaList = ser.ReadObject(ms) as List<object>;
+                ms.Close();
+                success = true;
+                return schemaList;
+            }
+            catch
+            {
+                success = false;
+                return new List<object>();
+            }
+        }
+        private Task<Tuple<bool, IList<string>>> Validate(List<object> schemas, object rules)
```

Comment:
>Validate [](http://example.com/codeflow?start=49&length=8)

You may want to rework the Validate method from the other comments. I haven't thought through, but you'll be dealing with serialized and deserialized ForgeTrees and Dictionary<string,ForgeTree>.

Since you need to the serialized string to parse into a JObject, consider serializing the ForgeTree/Dictionary<string, ForgeTree> before calling into Validate.

The Validate method should become more straightforward, just having to parse the string schema and check if it's valid against the rules.